tbdex/quote/quote.go

```diff
@@ -223,17 +231,24 @@ func DetailsInstruction(p *PaymentInstruction) QuoteDetailsOption {
 	}
 }
 
-// NewQuoteDetails creates a [QuoteDetails] object with the specified currency code, amount,
-// and optional modifications provided through [QuoteDetailsOption] functions.
-func NewQuoteDetails(currencyCode string, amount string, opts ...QuoteDetailsOption) QuoteDetails {
+// NewQuoteDetails creates a [QuoteDetails] object with the specified currency code, subtotal,
+// and optional modifications provided through [QuoteDetailsOption] functions, such as [DetailsFee] or [DetailsInstruction].
+func NewQuoteDetails(currencyCode string, subtotal decimal.Decimal, opts ...QuoteDetailsOption) QuoteDetails {
 	q := quoteDetailsOptions{}
 	for _, opt := range opts {
 		opt(&q)
 	}
+
+	total := subtotal
+	if !q.Fee.IsZero() {
+		total = subtotal.Add(q.Fee)
+	}
```

Comment:
Could do this. It works out the same whether Fee is zero or not
```suggestion
	total := subtotal.Add(q.Fee)
```

Reply:
tested and done! thanks https://github.com/TBD54566975/tbdex-go/pull/53/commits/173ddb057b2c2f6d8ba21216b79fa3c67fc52116